### ISSUE TITLE
Warning of XML not valid according to schematron but code still compiles

### ISSUE THREAD

**Issue by sfregoso** [nasa/fprime#325]
| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|   |
---
## Problem Description
When running the 'fprime-util impl' on my component, I get the following warning but the code still compiles 100%.

WARNING: XML file CameraManagerComponentAi.xml is not valid according to schematron <path_hidden_by_me>/Autocoders/Python/schema/default/active_comp_schematron.rng.

## How to Reproduce

1. 
2.
3.

## Expected Behavior

A description of the expected behavior.

**Comment by billallen256** [comment]
For some reason an `FprimeXmlException` is not raised in this `elif` block, unlike the `if` block above it:

https://github.com/nasa/fprime/blob/devel/Autocoders/Python/src/fprime_ac/parsers/XmlComponentParser.py#L1235

**Comment by LeStarch** [comment]
This is likely an oversight.

**Comment by billallen256** [comment]
I took a stab at this and added the `FprimeXmlException` for schematron, but then some of the tests wouldn't pass any more because they were defined as `active` components, but without an `async_input` port.

In most cases I was able to change the `sync_input` to `async_input`.  However, the thorniest one was `interface1`, which has no ports at all:
https://github.com/nasa/fprime/blob/devel/Autocoders/Python/test/interface1/TestComponentAi.xml

I tried hacking in a single `async_input` port for `interface1`, but then started to get linker errors.
